File: dagger_app/src/main/java/my/liewjuntung/daggerviperusage/app/ViperUsageApplication.java
package my.liewjuntung.daggerviperusage.app;

import android.app.Activity;
import android.app.Application;

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.HasActivityInjector;
import my.liewjuntung.daggerviperusage.dagger.DaggerAppComponent;

/**
 * Created by toruchoi on 06/10/2017.
 */

public class ViperUsageApplication extends Application implements HasActivityInjector {
    private static final String TAG = ViperUsageApplication.class.getSimpleName();

    @Inject
    DispatchingAndroidInjector<Activity> androidInjector;

    @Override
    public void onCreate() {
        super.onCreate();
        DaggerAppComponent
                .builder()
                .application(this)
                .build()
                .inject(this);
    }

    @Override
    public AndroidInjector<Activity> activityInjector() {
        return androidInjector;
    }
}

File: kotlin_dagger_app/src/main/java/my/liewjuntung/kotlindaggerapp/app/ViperUsageApplication.kt
package my.liewjuntung.kotlindaggerapp.app

import android.app.Activity
import android.app.Application
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector
import my.liewjuntung.daggerviperusage.dagger.DaggerAppComponent
import javax.inject.Inject

/**
 * Created by pandawarrior on 24/10/2017.
 */
class ViperUsageApplication: Application(), HasActivityInjector {
    @Inject
    lateinit var androidInjector: DispatchingAndroidInjector<Activity>

    override fun activityInjector(): AndroidInjector<Activity> {
        return androidInjector
    }

    override fun onCreate() {
        super.onCreate()
        DaggerAppComponent
                .builder()
                .application(this)
                .build()
                .inject(this)
    }
}